### PR TITLE
Connect modularization

### PR DESCRIPTION
**Pull request by marekrjpolak**
# Description

This PR has two separate parts:

In the first one, dynamic imports are used and build processes changed in `@trezor/connect` so that shitcoin-related code is in separate chunks and loaded on demand. Thanks to that, users should be better protected against hypothetical side effects or vulnerabilities in 3rd party code used by shitcoins they don't use.

In the second part, cardano stuff (mostly transaction composition) including `@fivebinaries/coin-selection` dependency is moved from suite to connect, thanks to mechanism introduced in 1) :point_up:

# Commits

## Stage 1

- https://github.com/trezor/trezor-suite/pull/9552/commits/5c626cecd2304caa228ceed2ebb23ef7251854ac - webpack added to `@trezor/suite-desktop-core`, so `@trezor/connect` dependency is now built separately into `/dist/connect` folder, including dynamically loaded chunks
- https://github.com/trezor/trezor-suite/pull/9552/commits/28020b10ba99a08051f279e369a7ae4366c5181a - blockchain-link workers are now loaded dynamically in node, so chunking is used for them
- https://github.com/trezor/trezor-suite/pull/9552/commits/8e2f8765611d261e4ae3a5ecf47541fcb616b29b - `getMethod` method in connect is now async and prepared for dynamic loading of shitcoin methods
- https://github.com/trezor/trezor-suite/pull/9552/commits/c330d3ce2a18b4325d54fc4b7411e6f5e4c0cec5, https://github.com/trezor/trezor-suite/pull/9552/commits/7dcee9749cbd3bf7c991c3ffd1151fe9b4eea649, https://github.com/trezor/trezor-suite/pull/9552/commits/23355fd12e9a3fe51b9b8ebf473316faca947135, https://github.com/trezor/trezor-suite/pull/9552/commits/60edc204088a9a57913a4023831311d66bfcddbb, https://github.com/trezor/trezor-suite/pull/9552/commits/ed7c502770b269f980975438b6b3d36f612c0af8, https://github.com/trezor/trezor-suite/pull/9552/commits/d0da0d03015b35054ae5d057cf6a5ad6b10a5142, https://github.com/trezor/trezor-suite/pull/9552/commits/fb2e18214083bd8096f44af0dd8e596d8bca4756, https://github.com/trezor/trezor-suite/pull/9552/commits/4963bd5ee5104daef186ecc89cd926f253fbb2f0 - shitcoin-related connect methods are now moved one by one to corresponding `api/coin/api` folders with their own index files (= entry points of dynamically loaded chunks)

## Stage 2

- https://github.com/trezor/trezor-suite/pull/9552/commits/9067c82ce9ea44942810771f5dbefd8a7dff74d2 - refactoring of cardano code in Suite, in order to create separate `composeTransaction` method and later move `@fivebinaries/coin-selection` dependency to Connect
- https://github.com/trezor/trezor-suite/pull/9552/commits/bfbfb13ddd73b9f051a0a51584ef6ce6790fb0d7 - create (still in Suite for now) new `cardanoComposeTransaction` method, enhanced `cardanoSignTransaction` method and `cardanoConnectUtils` helper module, all of which should later be moved to Connect
- https://github.com/trezor/trezor-suite/pull/9552/commits/df97ac9b31e257ed731bd4bad95158215d277fd0 - Cardano tx types in Suite polished, so they can be later moved to Connect as well
- https://github.com/trezor/trezor-suite/pull/9552/commits/672ef6a6c6be972bb2c4e5a144346ce05c32db8a - `cardanoComposeTransaction`, `cardanoConnectUtils` and Cardano tx types moved to Connect
- https://github.com/trezor/trezor-suite/pull/9552/commits/6e20664dc67c42ab34c9db6c81b6a7dd8d4cc047 - backwards-compatible enhancement of `cardanoSignTransaction` in Connect
- https://github.com/trezor/trezor-suite/pull/9552/commits/0e909c231e8030e39b7125309e3a564b9fd745a7 - some of cardano utils unit tests moved to Connect
- https://github.com/trezor/trezor-suite/pull/9552/commits/6466e203841020d56698ee31b23db9e4321f3ba8 - replace temporary pseudoconnect method with the real Connect ones, remove unneeded code
- https://github.com/trezor/trezor-suite/pull/9552/commits/f8566461edfc67dc81dcf475cb777b898e90ec5e - bump `@fivebinaries/coin-selection` version, so [node version](https://github.com/fivebinaries/coin-selection/pull/27) of code is used by default (needed for node Connect) with overload for browsers
- https://github.com/trezor/trezor-suite/pull/9552/commits/3186b19248dea5bc56a519ee35b25558282203b9 - set `asyncWebAssembly` flags and aliases needed for correctly including wasm files in the final build
- https://github.com/trezor/trezor-suite/pull/9552/commits/bb0b107386c106981da2595be626e48d562c9db9 - just some random unit test in Suite to reach required coverage

# What to test
First, we have to ensure that Connect as such works with the new build processes in all the supported environments (e.g. dev/prod, web/desktop/native, node/iframe, with or without popup, ...).

Then, mainly Cardano code has to be tested. Most changes happened in tx composition (so send form mostly) and in staking (compose method is used there as well, together with enhanced sign method). Both happy paths and errors should be exactly same as before.

### CODE REVIEW

**Review by mroz22 — APPROVED** [review]
very nice. I believe this is a huge step forward

**Review by Nodonisko — APPROVED** [review]

File: packages/blockchain-link/src/index.ts [hunk 1]
```diff
@@ -8,17 +8,17 @@ import type * as ResponseTypes from '@trezor/blockchain-link-types/lib/responses
 import type * as MessageTypes from '@trezor/blockchain-link-types/lib/messages';
 import type { Events } from '@trezor/blockchain-link-types/lib/events';
 
-const workerWrapper = (factory: BlockchainSettings['worker']): Worker => {
+const workerWrapper = (factory: BlockchainSettings['worker']): Worker | Promise<Worker> => {
```
Comment (Nodonisko) [inline packages/blockchain-link/src/index.ts:11]: What's the reason for these changes?

**Comment by mroz22** [comment]
lets not merge this before freeze wich is Oct 31

File: packages/blockchain-link/src/index.ts (hunk 1 above)
Comment (marekrjpolak) [reply packages/blockchain-link/src/index.ts:11]: That worker code is now chunked and loaded dynamically in desktop app: https://github.com/trezor/trezor-suite/pull/9552/files#diff-7716251d3a5873ffcd11f408f34b57a364fb2cdabaa1c4fb433c1ce52a5284cd

**Review by mroz22 — APPROVED** [review]
